pkg/promclient/promclient.go

```diff
@@ -97,14 +97,17 @@ func NewWithTracingClient(logger log.Logger, userAgent string) *Client {
 	)
 }
 
-func (c *Client) get2xx(ctx context.Context, u *url.URL) (_ []byte, _ int, err error) {
+func (c *Client) get2xx(ctx context.Context, u *url.URL, additionalHeaders map[string]string) (_ []byte, _ int, err error) {
```

Comment:
Wonder if `headers ...[2]string` wouldn't do (:  WDYT?

Reply:
Umm, I don't have a strong preference here. But seems using [2]string is better than a map because it uses less memory.

Reply:
Let me update the pr.